testbeam_analysis/track_analysis.py

```diff
@@ -984,10 +1001,14 @@ def _kalman_fit_3d(hits, dut_fit_selection, transition_matrix, transition_covari
     x_err = np.sqrt(np.diagonal(cov, axis1=3, axis2=2))[:, :, 0]
     y_err = np.sqrt(np.diagonal(cov, axis1=3, axis2=2))[:, :, 1]
 
+    # Check for invalid values (NaN)
+    if np.any(np.isnan(smoothed_state_estimates)):
+        logging.warning('Not all smoothed state estimates have valid values (Array contains NANs)! Check input of  Kalman Filter.')
+
     return smoothed_state_estimates, chi2, x_err, y_err
 
 
-def _fit_tracks_kalman_loop(track_hits, dut_fit_selection, pixel_size, z_positions, beam_energy, total_thickness, radiation_length):
+def _fit_tracks_kalman_loop(track_hits, dut_fit_selection, pixel_size, n_pixels, z_positions, beam_energy, total_thickness, radiation_length, add_scattering_center=None):
```

Comment:
Maybe no std. parameter for `add_scattering_center` to not hide possible bugs when using the function?